delivery_internal/models/stock_pickup_request.py

```diff
@@ -0,0 +1,141 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 LasLabs Inc.
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html).
+
+from odoo import api, fields, models
+
+
+class StockPickupRequest(models.Model):
+    _name = 'stock.pickup.request'
+    _description = 'Stock Pickup Request'
+
+    name = fields.Char(
+        string='Pickup Name',
+        default=lambda s: s._default_name(),
+    )
+    company_id = fields.Many2one(
+        string='Responsible Company',
+        comodel_name='res.company',
+        required=True,
+    )
+    cash_on_delivery = fields.Boolean()
+    state = fields.Selection(
+        selection=[
+            ('new', 'New'),
+            ('confirmed', 'Confirmed'),
+            ('done', 'done')
+        ],
+        compute='_compute_state',
+        store=True,
+    )
+    picking_id = fields.Many2one(
+        string='Source Picking',
+        comodel_name='stock.picking',
+        required=True,
+    )
+    in_picking_id = fields.Many2one(
+        string='Inbound Buyer Picking',
+        comodel_name='stock.picking',
+    )
+    out_picking_id = fields.Many2one(
+        string='Carrier Pickup Picking',
+        comodel_name='stock.picking',
+    )
+    cash_in_picking_id = fields.Many2one(
+        string='Cash Picking From Buyer Picking',
+        comodel_name='stock.picking',
+    )
+    cash_out_picking_id = fields.Many2one(
+        string='Cash Delivery From Carrier Picking',
+        comodel_name='stock.picking',
+    )
+
+    def _get_pick_type(self, picking, op_type):
+        """ It returns the picking type for the given picking's carrier
+        Args:
+            picking (stock.picking): The parent picking to use
+            op_type (str): Whether we need to stock or cash pick type
+        Returns:
+            stock.picking.type: The picking type
+        """
+        return getattr(
+            picking.carrier_id,
+            'internal_delivery_%s_picking_type_id' % op_type
+        )
+
+    @api.multi
+    @api.depends('in_picking_id.state', 'in_picking_id.state')
+    def _compute_state(self):
+        """ It will compute the status of the pickup based on the delivery
+         pickings
+         """
+        for rec in self:
+            state = 'new'
+            in_state = rec.in_picking_id.state
+            out_state = rec.out_picking_id.state
+            if in_state in ('confirmed', 'done') and in_state == out_state:
+                state = in_state
```

Comment:
Fixed! This also uncovered a hole in the RFP module logic, which was subsequently fixed.